src/core/MOM.F90

```diff
@@ -1627,7 +1627,8 @@ subroutine step_MOM_thermo(CS, G, GV, US, u, v, h, tv, fluxes, dtdia, &
           call remap_dyn_split_RK2_aux_vars(G, GV, CS%dyn_split_RK2_CSp, h, h_new, CS%ALE_CSp, CS%OBC, dzRegrid)
 
         if (associated(CS%OBC)) &
-          call pass_var(h_new, G%Domain)
+          call pass_var(h, G%Domain, halo=max(2,CS%cont_stencil), complete=.false.)
```

Comment:
There is a problem here with the logical branch of the code that is enabled when `CS%OBC` is associated.  There is no `then` and `endif` for the logical test, so it is just the next line that is exercised.  From the intending, one might expect (incorrectly?) that all 3 of these lines are intended to be called only when `associated(CS%OBC)` is true.  With this change, `call  pass_var(h_new, ...)` always occurs, as does the call to `remap_OBC_fields()`.  Looking into the latter call, it is not too problematic because it returns right away when CS%OBC is not associated.

  However, I think that this block of code should be reexamined to ensure that the logic matches what is actually intended, and then that the indenting matches what the logic does.